### ISSUE TITLE
Better handling of exceptions

### ISSUE THREAD

**Issue by ducasp** [Fubukimaru/MSXIRC#4]
Program is MSX2 or better only, so it should not try to execute on MSX1 and warn about it.

Also, if there is no TCPIP UNAPI available, should just quite and leave an error message about it.

It is possible to exit the program hitting CTRL+C on several places, and it won't return the screen to regular fonts / regular colors and functionality

Program need at least a few mapper segments free and will run without mapper and crash when connecting

**Comment by ducasp** [comment]
So far what has been done:

MSX2 or better check, otherwise return with error message -> Done

Working now on the other three exceptions :)

**Comment by ducasp** [comment]
I've implemented the UNAPI checking early in the program proccess, thus, allowing the program to exit before changing the screen mode and leaving a message that will be visible on MSX-DOS.

The same has been done on the mapper initialization routines, done earlier (even before UNAPI checking), so if no mapper is found or if we don't have at least 2 free segments, it exits telling that it needs at least 32KB free from the memory mapper.

Also, fixed the mapper routine that was causing the loss of the FF segment when on DOS-1, and also, was prone to overwrite the segment UNAPI is installed if using a memory mapper UNAPI. I think that the original author confused the UNAPI specification, it says that a memory mapped UNAPI MUST NOT use FF mapper segment, because returning FF segment means UNAPI is in a ROM slot not in memory mapper. Nestor made this choice for UNAPI because nowadays most people use DOS2 and DOS2 allocates FF segment for its own usage. 

Bug 1-> MSXIRC was reserving FF segment always on MSX DOS 1 (on DOS2, DOS2 handles that and there is no error), thinking that the ram mapped UNAPI would be installed on it. It is very much the opposite, it will be on any other segment that is not FF. So, it was very possible that on MSX DOS 1, when opening a few windows, the UNAPI segment would be overwritten with a WCB. Fixed so it no longer avoid using FF segment, that is specially missed on a 128KB machine running a rom UNAPI :)

Bug 2 -> When a ram mapped UNAPI was used on DOS 1, the segment it was sitting on would be marked as FREE for use (as it is, remember, on any segment other than FF). Now, when using DOS 1, we mark the real segment the UNAPI MAPPER is as used on our internal mapper allocation table, so MSXIRC will no longer overwrite it with a WCB when opening multiple windows
